partiql-lang/src/main/kotlin/org/partiql/lang/eval/ExprFunction.kt

```diff
@@ -45,6 +48,18 @@ interface ExprFunction {
     fun callWithRequired(session: EvaluationSession, required: List<ExprValue>): ExprValue {
         errNoContext("Invalid implementation for ${signature.name}#call", ErrorCode.INTERNAL_ERROR, true)
     }
+
+    @Deprecated("This function has been deprecated", level = DeprecationLevel.ERROR)
```

Comment:
Would be useful to provide information regarding function overloading. Something like:
```
As function overloading is now supported, please provide multiple implementations of ExprFunction.
```